docs/how-to-guides/run-chainhook-as-a-service-using-bitcoind.md

```diff
@@ -0,0 +1,323 @@
+---
+title: Run Chainhook as a service using Bitcoind
+---
+
+# Run Chainhook as a service using Bitcoind
+
+Chainhoook node can interact with Bitcoin and Stacks blockchains. However, chainhook can get blocks from either bitcoind or Stacks. This document helps you with running chainhook as a service using Bitcoind. To run chainhook as a service using Stacks, refer to [this](run-chainhook-as-a-service-using-stacks.md).
+
+## Prerequisites
+
+- Install bitcoind using [this](https://bitcoin.org/en/bitcoin-core/) link.
+- Install the latest version of chainhook by following [Install Chainhooks from source](../getting-started.md#install-chainhooks-from-source).
+
+Bitcoind installation will download binaries in a zip format, `bitcoin-22.0-osx64.tar.gz`. You can extract the zip file to view the folders. Expand the `bin` folder to see executable files. The `bitcoind` executable file also exists here.
+
+## Steps to run chainhooks as a service
+
+To run chainhook as a service, you will follow the steps:
+
+- Ingesting blocks through bitcoind
+  - To understand predicates for bitcoind, refer to [how to use chainhook with bitcoind](how-to-use-chainhook-with-bitcoin.md)
+- Test your predicates
+- Deploy your predicates
+
+## Ingesting blocks through bitcoind
+
+Ingesting blocks happens through zeromq, an embedded networking library in the bitcoind installation package.
+
+### Prepare the bitcoind node
+
+- Create a config file `bitcoin.conf` on your local machine, use the configurations below and save it.
+- Get the downloaded path of the Bitcoind from the [prerequisites section](#prerequisites) and use it in the `datadir` configuration below.
+- Set a username of your choice for bitcoind and use it in the `rpcuser` configuration below.
+- Set a password of your choice for bitcoind and use it in the `rpcpassword` configuration below.
+
+>[!NOTE]
+> Make a note of the `rpcuser`, `rpcpassword` and `rpcport` values to use later in the chainhook configuration.
+
+
+```conf
+datadir=<path-to-your-downloaded-bitcoind>
+server=1
+rpcuser="bitcoind_username"  --> You can set the username here
+rpcpassword="bitcoind_password"  --> You can set the password here
+rpcport=8332   --> You can set your localhost port number here
+rpcallowip=0.0.0.0/0
+rpcallowip=::/0
+txindex=1
+listen=1
+discover=0
+dns=0
+dnsseed=0
+listenonion=0
+rpcserialversion=1
+disablewallet=0
+fallbackfee=0.00001
+rpcthreads=8
+blocksonly=1
+dbcache=4096
+
+# Start zeromq
+zmqpubhashblock=tcp://0.0.0.0:18543
+```
+
+### Run the bitcoind node
+
+Now that you have `bitcoin.conf` file ready with the bitcoind configurations, you can run the bitcoind node.
+In the command below, use the path to your `bitcoin.conf` file from your machine and run the command in the terminal.
+
+```bash
+$ ./bitcoind -conf=<path-to-bitcoin.config>/bitcoin.conf
```

Comment:
Since this command is starting a bitcoin node, there's going to be a pretty huge initial sync. It may be worth mentioning that this startup process takes a few hours